src/umpire/strategy/MixedPool.hpp

```diff
@@ -0,0 +1,72 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_MixedPool_HPP
+#define UMPIRE_MixedPool_HPP
+
+#include <memory>
+#include <vector>
+#include <map>
+
+#include "umpire/strategy/AllocationStrategy.hpp"
+
+#include "umpire/strategy/DynamicPool.hpp"
+#include "umpire/strategy/FixedPool.hpp"
+
+#include "umpire/Allocator.hpp"
+
+namespace umpire {
+namespace strategy {
+
+/**
+ * \brief A faster pool that pulls from a series of pools
+ *
+ * Pool implementation using a series of FixedPools for small sizes,
+ * and a DynamicPool for sizes larger than (1 << LastFixed) bytes.
+ */
+class MixedPool :
+  public AllocationStrategy
+{
+  public:
+    MixedPool(const std::string& name, int id,
```

Comment:
I think we need a way to also specify options for the internal DynamicPool.

Reply:
Yes, that's a good idea. New commit adds those.